File: 3/sol1.cpp
#include <iostream>
#include <fstream>
#include <cstdio>
#include <unordered_map>
#include <unordered_set>


int main() {
	int count = 0;
	std::ifstream in("input");
	std::string line;
	std::unordered_map<int, std::unordered_set<int> > covered;
	while(std::getline(in, line)){
		int left, top, width, height;
		sscanf(line.c_str(), "%*s @ %d,%d: %dx%d", &left, &top, &width, &height);
		for(int i=left; i<left+width; i++){
			for(int j=top; j<top+height; j++){
				if(covered[i].count(j)){
					count++;
				} else {
					covered[i].insert(j);
				}
			}
		}
	}

	std::cout << "count: " << count << std::endl;
	return 0;
}

File: 3/sol1.py
import re
from collections import defaultdict


covered = defaultdict(int)
for parsed in map(lambda i: i.split('@')[1].strip(), open('input')):
    left, top, width, height = map(int, re.sub(r'[,:x]', ' ', parsed).split())
    for x in range(left, left + width):
        for y in range(top, top + height):
            covered[(x, y)] += 1

print(len([k for k, v in covered.items() if v > 1]))
